StatePrinter/Introspection/IntroSpector.cs

```diff
@@ -201,15 +203,14 @@ private bool IntrospectIEnumerable(object source, Field field)
             seenBefore.TryGetValue(source, out optionReferenceInfo);
 
             tokens.Add(new Token(TokenType.FieldnameWithTypeAndReference, field, null, optionReferenceInfo, source.GetType()));
-            tokens.Add(StartEnumeration);
+            tokens.Add(StartList);
 
-            int i = 0;
+            long i = 0;
```

Comment:
Sorry, force of habit. In my day job the assumption of fewer than 2B elements per sequence doesn't hold.

I reverted it back to int. There are probably a few more places where StatePrinter won't handle huge sequences so there is no point in using long just in this place.